### PR TITLE
Directive restrict EA. Dynamic Options change.

### PR DESCRIPTION
**Pull request by hlege**
Added the abilitiy to be able to define the directive in attribute instead of element. Replace scope.$watch() to scope.$watchCollection() to be able to refresh dynamic changes in the option list like array.push(...)

### CODE REVIEW

File: src/select2.js [hunk 1]
```diff
@@ -46,12 +46,14 @@ angular.module("rt.select2", [])
...
-            template: "<input type=\"hidden\"></input>",
+            restrict: "EA",
+            template: "<input type=\"hidden\"/>",
             replace: true,
             link: function (scope, element, attrs, controller) {
                 var getOptions;
-
+                element.removeAttr("select2");
+                element.removeAttr("data-select2");
+                element.removeData("select2");
```
Comment (rubenv) [inline src/select2.js]: Why do this?

**Comment by Zenorbi** [comment]
Those lines are needed because select2 uses jQuery's `element.data("select2")` to store its data, but also refuses to initialize if the returned value is anything other than `undefined` (already initialized).

If the directive is initialized using `data-select2=""`, jQuery loads the `""` value to `element.data("select2")`, so the select2 won't initialize. The `element.removeData("select2")` will clear that data value, but will try to read the attribute again if the attribute is left on the element.

I don't know why the `element.removeAttr("select2")` is needed.

**Comment by hlege** [comment]
`element.removeAttr("select2")` is not necessary. It was a mistake.

**Comment by rubenv** [comment]
So it's a bunch of workaround for allowing `A`. Why would you even want that? The element gets replaced anyway, so only allowing `E` is probably the right way to go here.

**Comment by Zenorbi** [comment]
Allowing only `E` would make the page non W3C compliant and also there is no guarantee that browsers will correctly include that custom element in the DOM (IE8).

**Comment by rubenv** [comment]
I'm sorry, but I have no interest in supporting IE8 in this library. That thing should just go away.

You're already risking that problem anyway if you add `required` to any input: it's not supported (and thus "not W3C compliant") by IE8. Yet it's the only way to trigger the "required" validator in Angular.JS. Turns out that works just fine, so the issue doesn't really exist.

If you really need IE8 support, just create a custom element similar to how [html5shiv](https://github.com/aFarkas/html5shiv) does it. It's time we stop including hacks for things that stopped being relevant ages ago.

**Pull request closed without merging**